### ISSUE TITLE
[Feature request] Export as JS library

### ISSUE THREAD

**Issue by KaKi87** [zelark/nano-id-cc#35]
For example, the following function :

```js
getCollisionProbability({
    alphabet: String,
    length: Integer,
    speed: Integer,
    speedUnit: String // 'hour' or 'second'
})
```

Could return the following value :

```js
{
    duration: Integer,
    durationUnit: String, // 'second', 'minute', 'hour', 'day', 'year'
    count: Integer
}
```

Thanks

**Comment by zelark** [comment]
Good idea. However, this project was planned as a simple tool, firstly just for fun, so I'm not interested in overcomplicating it.

**Comment by KaKi87** [comment]
Why would that be overcomplicating anything ?

As far as I know, creating APIs is even less complicated than creating GUIs.

**Comment by zelark** [comment]
Because you have to create a library, then require it as a dependency. Anyway, I'm not interested in it.

**Comment by KaKi87** [comment]
Okay, how about at least converting those `cljs` files into `js` files so that I could create an exported function myself ?

Thanks

**Comment by zelark** [comment]
There is JS version https://github.com/Alex7Kom/nano-nanoid-cc, I guess you can use code from there.

**Comment by KaKi87** [comment]
Yep, all its functions are here : https://github.com/alex7kom/nano-nanoid-cc/blob/92bc88355dfe512a6d72a634b8daf77817ff87cc/src/util/calc.js

However, despite containing the time variable, it doesn't contain the count variable.

Thanks

**Comment by KaKi87** [comment]
Could you help me get this variable ?

Thanks

**Comment by JacobWeisenburger** [comment]
@KaKi87
What do you mean by "the count variable"? Do you mean the ID length?

**Comment by KaKi87** [comment]
No, I meant `critical-number`. Thanks

**Comment by JacobWeisenburger** [comment]
I believe it's this:
https://github.com/alex7kom/nano-nanoid-cc/blob/92bc88355dfe512a6d72a634b8daf77817ff87cc/src/util/calc.js#L7C1-L12C1

**Comment by KaKi87** [comment]
That seems right :

```js
import {
    getRandomBits,
    getGenerateForCollision
} from 'https://esm.sh/gh/alex7kom/nano-nanoid-cc@92bc88355d/src/util/calc.js';

const
    alphabet = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz-',
    length = 21,
    probabilityPercent = 1,
    criticalNumber = getGenerateForCollision(getRandomBits(alphabet.length, length), probabilityPercent / 100),
    { format } = Intl.NumberFormat('en', { notation: 'compact' });

console.log(`${format(criticalNumber)} IDs needed for ${probabilityPercent}% collision probability.`);
// -> 1,307,661T IDs needed for 1% collision probability.
```

Although, @alex7kom's number is higher than yours ? :thinking: 

Thanks

**Comment by zelark** [comment]
Now the calculator is rewritten in TypeScript, so I'm going to close this issue.

See https://github.com/zelark/nano-id-cc/pull/37

**Comment by KaKi87** [comment]
So what's the way to do this now ? The PR doesn't really say. Thanks

**Comment by zelark** [comment]
Hi @KaKi87. Publishing this as a JS/npm library is out of scope for this project — it's intentionally a single-purpose tool (so the `wont be done` label). So there's no `getCollisionProbability` package to install.

That said, after the rewrite the math is a set of plain, dependency-free TypeScript functions you can lift into your own project:

https://github.com/zelark/nano-id-cc/blob/master/src/calc.ts

```TypeScript
import { randomBits, criticalNumber, timeToCollision } from './calc';

const alphabet = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz-';
const length = 21;
const probability = 0.01;        // 1% collision probability
const speedPerSecond = 1000 / 3600; // e.g. 1000 IDs per hour

const bits = randomBits(alphabet.length, length);
const count = criticalNumber(bits, probability); // this is the "count"/critical-number
const seconds = timeToCollision(count, speedPerSecond);
```

A couple of notes:

`criticalNumber` is the value you were after as "count" / `critical-number`.
Pass the number of unique characters in the alphabet (dedupe first), and a speed in IDs per second — that's how `src/format.ts` uses these to build the UI output.

Since the file is self-contained, just copy it (or vendor the few lines you need) rather than expecting an installable package.

**Comment by KaKi87** [comment]
Well, the old code still works anyway, and if necessary, the imports can be migrated to this repo without changing the rest of the code, as follows :

```js
import {
    randomBits as getRandomBits,
    criticalNumber as getGenerateForCollision
} from 'https://esm.sh/gh/zelark/nano-id-cc@fd7f97260a/src/calc.ts';
```

Thanks anyway
